WcaOnRails/lib/solve_time.rb

```diff
@@ -1,21 +1,65 @@
 class SolveTime
+  EMPTY_STRING = ''.freeze
+  CLOCK_FORMAT = "%d:%02d:%02d.%02d".freeze
+  DOT_STRING = ".".freeze
+  ZERO_STRING = "0".freeze
+  DNF_STRING = "DNF".freeze
+  DNS_STRING = "DNS".freeze
+  QUESTION_STRING = "?:??:??".freeze
+
+  include Comparable
+
+  attr_reader :wca_value
   def initialize(event_id, field, wca_value)
+    raise "Unrecognized wca_value: #{wca_value}" if wca_value < -2
     @event_id = event_id
     @field = field
     @wca_value = wca_value
   end
 
+  DNF_VALUE = -1
+  DNF = SolveTime.new(nil, nil, DNF_VALUE)
+  DNS_VALUE = -2
+  DNS = SolveTime.new(nil, nil, DNS_VALUE)
+  SKIPPED_VALUE = 0
+  SKIPPED = SolveTime.new(nil, nil, SKIPPED_VALUE)
+
+  def dn?
+    dnf? || dns?
+  end
+
+  def dns?
+    wca_value == DNS_VALUE
+  end
+
+  def dnf?
+    wca_value == DNF_VALUE
+  end
+
+  def skipped?
+    wca_value == SKIPPED_VALUE
+  end
+
+  def to_orderable
```

Comment:
I've fixed this, but I had to keep the `dnf?` case in here since DNF is represented as `-1` which would lead to `DNF < solve_time(30)`.